modules/programs/zsh.nix

```diff
@@ -641,7 +641,18 @@ in
 
         (optionalString cfg.autosuggestion.enable ''
           source ${pkgs.zsh-autosuggestions}/share/zsh-autosuggestions/zsh-autosuggestions.zsh
-          ZSH_AUTOSUGGEST_STRATEGY=(${concatStringsSep " " cfg.autosuggestion.strategy})
+          ${optionalString (cfg.autosuggestion.strategy != []) ''
+            # Make strategy configuration optional in case the user wants absolutely
+            # full control with when and where zsh-autosuggestions' strategy variable
+            # is placed in the zshrc
+            if [ -n "''${ZSH_AUTOSUGGEST_STRATEGY:-}" ]; then
+              typeset -U ZSH_AUTOSUGGEST_STRATEGY # Remove any duplicate entries
```

Comment:
We could use lib.unique for it.

Reply:
>Because I notice that even the zsh autosuggestion plugin itself actually creates the ZSH_AUTOSUGGEST_STRATEGY variable itself, which adds the duplicate variables if you simply want to append the variable, why I had to add `typeset -U`

This one here in particular I was referring to: https://github.com/zsh-users/zsh-autosuggestions/blob/0e810e5afa27acbd074398eefbe28d13005dbc15/src/config.zsh#L18-L21

Also now I'm considering taking out the entire boolean check altogether since the plugin itself adds the variable in the first place and just leave it as simply this 🤔 
```
typeset -U ZSH_AUTOSUGGEST_STRATEGY # Remove any duplicate entries
ZSH_AUTOSUGGEST_STRATEGY=(${concatStringsSep " " cfg.autosuggestion.strategy} "''${ZSH_AUTOSUGGEST_STRATEGY[@]}")
```